### ISSUE TITLE
[Suggest] About a type of parameter in LossFunction(BCE,BCEWithLogits)

### ISSUE THREAD

**Issue by dohwan2** [pytorch/pytorch#5800]
- OS: Windows7
- PyTorch version: 0.3.1
- How you installed PyTorch (conda, pip, source): conda
- Python version: 3.54
- CUDA/cuDNN version: None
- GPU models and configuration: None
- GCC version (if compiling from source): None

In addition, including the following information will also be very helpful for us to diagnose the problem:
- I recommend changing the type of the parameter(BCELoss, BCEWithLogits)
- When users use nn.BCELoss(), the type of 'weight' parameter is Variable
  -> example : loss = nn.BCELoss(weight=Variable(weight)) 
- When users use nn.BCEWithLogitsLoss(), the type of 'weight' parameter is Tensor
 -> example : loss = nn.BCEWithLogitsLoss(weight=torch.FloatTensor(weight))

I know why this difference happens.
In BCELoss
 -> return F.binary_cross_entropy(input, target, weight=self.weight, size_average=self.size_average)

In BCELossWithLogitsLoss
 -> return F.binary_cross_entropy_with_logits(input, target, Variable(self.weight), self.size_average)

I know it's not a critical problem, this is just my opinion.

**Comment by apaszke** [comment]
This is already resolved in master. Thanks for the report!
